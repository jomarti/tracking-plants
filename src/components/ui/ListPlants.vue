<template>
  <div>
    <h1 class="title font-bold text-xl mb-4">
      Mis plantas
    </h1>
    <ul>
      <li>
        <button class="border border-black text-3xl rounded-full py-1 px-3">
          +
        </button>
      </li>
    </ul>
  </div>
</template>

<script>
export default {
  name: 'ListPlants'
};
</script>
