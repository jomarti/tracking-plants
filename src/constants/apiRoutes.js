const PLANTS_API_BASE_ROUTE = 'https://trefle.io/api/v1/';
const PLANTS_TOKEN = `?token=${process.env.VUE_APP_PLANTS_API_TOKEN}`;

export const API_ROUTES = {
  ALL_PLANTS: `${PLANTS_API_BASE_ROUTE}plants/search${PLANTS_TOKEN}`
};
