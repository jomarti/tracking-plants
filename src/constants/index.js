import { API_ROUTES } from '@/constants/apiRoutes';
import { ROUTES } from '@/constants/routes';

export { ROUTES, API_ROUTES };
