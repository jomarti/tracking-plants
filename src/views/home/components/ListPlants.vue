<template>
  <div>
    <h1 class="title font-bold text-xl mb-4">
      Mis plantas
    </h1>
    <ul>
      <li>
        <button
          class="border border-black text-3xl rounded-full py-1 px-3"
          @click="goToCreatePlant"
        >
          +
        </button>
      </li>
    </ul>
  </div>
</template>

<script>
import { ROUTES } from '@/constants';
export default {
  name: 'ListPlants',
  methods: {
    goToCreatePlant() {
      this.$router.push({ name: ROUTES.ADD.name });
    }
  }
};
</script>
