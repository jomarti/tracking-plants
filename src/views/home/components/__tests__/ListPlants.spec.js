import { ROUTES } from '@/constants';
import { shallowMount } from '@vue/test-utils';
import ListPlants from '../ListPlants';

describe('HomeView -> ListPlants', () => {
  const $router = {
    push: jest.fn()
  };
  const build = () => {
    const wrapper = shallowMount(ListPlants, {
      mocks: {
        $router
      }
    });
    return {
      wrapper
    };
  };

  it('should match snapshot', () => {
    const { wrapper } = build();
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should go to add route and button is clicked', () => {
    const { wrapper } = build();
    const button = wrapper.find('[data-test-id="add-button"]');
    button.trigger('click');

    expect(wrapper.vm.$router.push).toHaveBeenCalledTimes(1);
    expect(wrapper.vm.$router.push).toHaveBeenCalledWith({
      name: ROUTES.ADD.name
    });
  });
});
