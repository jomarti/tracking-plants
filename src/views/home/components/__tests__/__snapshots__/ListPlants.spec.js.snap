// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView -> ListPlants should match snapshot 1`] = `
<div>
  <h1
    class="title font-bold text-xl mb-4"
  >
    
    Mis plantas
  
  </h1>
   
  <ul>
    <li>
      <button
        class="border border-black text-3xl rounded-full py-1 px-3"
        data-test-id="add-button"
      >
        
        +
      
      </button>
    </li>
  </ul>
</div>
`;
